tools/deploy/scripts/compose.sh

```diff
@@ -21,149 +21,255 @@ set -e
 # Without `PROD_DEPLOY` exported flag the script will use default arguments
 # If first arg is `multiple`, script will start 4 fluence node along with Swarm & Parity nodes
 
-# Update all containers
-docker pull parity/parity:v2.3.0
-docker pull ethdevops/swarm:edge
-docker pull fluencelabs/node
-docker pull fluencelabs/worker
-
 # `PROD_DEPLOY` variable is assigned in `fabfile.py`, so if run `compose.sh` directly,
 #  the network will be started in development mode locally
-if [ -z "$PROD_DEPLOY" ]; then
-    echo "Deploying locally with default arguments."
-    export NAME='fluence-node-1'
-    # open 10 ports, so it's possible to create 10 workers
-    export PORTS='25000:25010'
-    # eth address in `dev` mode Parity with eth
-    export OWNER_ADDRESS=0x00a329c0648769a73afac7f9381e08fb43dbea72
-    export PRIVATE_KEY=4d5db4107d237df6a3d58ee5f70ae63d73d7658d4026f2eefd2f204c81682cb7
-    export PARITY_ARGS='--config dev-insecure --jsonrpc-apis=all --jsonrpc-hosts=all --jsonrpc-cors="*" --unsafe-expose'
-else
-    echo "Deploying for $CHAIN chain."
-    export PARITY_ARGS='--light --chain '$CHAIN' --jsonrpc-apis=all --jsonrpc-hosts=all --jsonrpc-cors="*" --unsafe-expose'
-fi
 
-# getting docker ip address
-case "$(uname -s)" in
-   Darwin)
-     export DOCKER_IP=host.docker.internal
-     ;;
-
-   Linux)
-     export DOCKER_IP=$(ifconfig docker0 | grep 'inet ' | awk '{print $2}' | grep -Po "[0-9\.]+")
-     ;;
-esac
-
-# use exported external ip address or get it from OS
-# todo rewrite this
-if [ -z "$PROD_DEPLOY" ]; then
-    EXTERNAL_HOST_IP="127.0.0.1"
+# generates json with all arguments required for registration a node
+function generate_json()
+{
+    # printing the command in the last line to parse it from a control script
+    DATA="{\"node_ip\": \"$EXTERNAL_HOST_IP\",
+           \"tendermint_key\": \"$TENDERMINT_KEY\",
+           \"tendermint_node_id\": \"$TENDERMINT_NODE_ID\",
+           \"contract_address\": \"$CONTRACT_ADDRESS\",
+           \"account\": \"$OWNER_ADDRESS\",
+           \"start_port\": $START_PORT,
+           \"last_port\": $LAST_PORT}"
+    JSON=$(echo $DATA | paste -sd "" - | awk 'NF')
```

Comment:
`paste -sd "" -` doesn't seem to work on macOS. But `paste -sd "\0" -` works just fine.
```suggestion
    JSON=$(echo $DATA | paste -sd "\0" - | awk 'NF')
```

Reply:
What's the purpose of `awk 'NF'`?